### PR TITLE
[DvdSubPes] - Working on #1932.

### PR DESCRIPTION
**Pull request by ivandrofly**
Fix for https://github.com/SubtitleEdit/subtitleedit/issues/1932

Not tested with file provided which is +3GB (I have a very bad internet connection), @niksedk  Can you test it with this patch?

### CODE REVIEW

File: libse/TransportStream/DvbSubPes.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public DvbSubPes(byte[] buffer, int index)
 
             HeaderDataLength = buffer[index + 8];
 
-            if (buffer.Length < index + 9 + HeaderDataLength)
+            if (buffer.Length < index + 10 + HeaderDataLength)
```
Comment (ivandrofly) [inline libse/TransportStream/DvbSubPes.cs]: If we say 9 the buffer's length may be equal to 9 but in [line 67](https://github.com/SubtitleEdit/subtitleedit/pull/1934/files#diff-1eb1be2135f55969cc79857272e0c6a9R67) we are trying to access buffer from index 9.

File: libse/TransportStream/DvbSubPes.cs [hunk 2]
```diff
@@ -68,26 +68,26 @@ public DvbSubPes(byte[] buffer, int index)
...
-                PresentationTimestampDecodeTimestampFlags == Helper.B00000011)
-            {
-                PresentationTimestamp = (ulong)buffer[tempIndex + 4] >> 1;
-                PresentationTimestamp += (ulong)buffer[tempIndex + 3] << 7;
-                PresentationTimestamp += (ulong)(buffer[tempIndex + 2] & Helper.B11111110) << 14;
-                PresentationTimestamp += (ulong)buffer[tempIndex + 1] << 22;
-                PresentationTimestamp += (ulong)(buffer[tempIndex + 0] & Helper.B00001110) << 29;
-            }
-            if (PresentationTimestampDecodeTimestampFlags == Helper.B00000011)
+            if (index + 9 + 4 < buffer.Length)
```
Comment (ivandrofly) [inline libse/TransportStream/DvbSubPes.cs]: Make sure all the element in buffers will be in range.

**Comment by xylographe** [comment]
:+1: Resolves issue #1932 for provided file.

**Comment by ivandrofly** [comment]
> Resolves issue #1932 for provided file.

Nice :). I think it can be merged now /cc @niksedk

**Comment by niksedk** [comment]
:)

**Pull request merged**